test/topo/topo.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	utils "github.com/onosproject/onos-topo/test/utils"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+var log = logging.GetLogger("topo")
+
+// TestAddRemoveDevice adds devices to the storage, lists and checks that they are in database and removes devices from the storage
+func (s *TestSuite) TestAddRemoveDevice(t *testing.T) {
+
+	log.Debugf("Creating connection")
```

Comment:
You should use t.Log for printing debug statement in test. Remove log from test.